### PR TITLE
More complete nmap support

### PR DESCRIPTION
**Pull request by gpotter2**
Nmap:
- Add a bunch of versions hash. I tested them all
- add old dist versions, add support for building nmap<7.5.0
- config args

I disabled the auto update by default because it barely works.

### CODE REVIEW

**Comment by gpotter2** [comment]
@spackbot request review

File: var/spack/repos/builtin/packages/nmap/package.py [hunk 1]
```diff
@@ -13,7 +13,47 @@ class Nmap(AutotoolsPackage):
...
+        if '+ncat' not in self.spec:
+            args.append('--without-ncat')
+        if '+ndiff' not in self.spec:
+            args.append('--without-ndiff')
+        if '+nping' not in self.spec:
+            args.append('--without-nping')
+        if '+zenmap' not in self.spec:
+            args.append('--without-zenmap')
+        if '+nmapupdate' not in self.spec:
+            args.append('--without-nmap-update')
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/nmap/package.py]: For most of these options you can use a method of `AutotoolsPackage`:
```python
args += self.with_or_without('liblua')
```
which will add either `--with-liblua` or `--without-liblua` depending on the value of the variant.

**Review by alalazo — CHANGES_REQUESTED** [review]
Basically LGTM, left a minor comment on the configure argument method.

File: var/spack/repos/builtin/packages/nmap/package.py (hunk 1 above)
Comment (alalazo) [reply var/spack/repos/builtin/packages/nmap/package.py]: @gpotter2

Comment (gpotter2) [reply var/spack/repos/builtin/packages/nmap/package.py]: Hi, sorry for the delay. I've updated the PR !

**Review by alalazo — APPROVED** [review]

**Pull request merged**